internal/lsp/cache/cache.go

```diff
@@ -180,6 +189,53 @@ func (c *Cache) SetModule(fileURI string, module *ast.Module) {
 	c.modules[fileURI] = module
 }
 
+func (c *Cache) SetFileAggregates(fileURI string, data map[string][]report.Aggregate) {
+	c.aggregateDataMu.Lock()
+	defer c.aggregateDataMu.Unlock()
+
+	flattenedAggregates := make([]report.Aggregate, 0)
+
+	for _, aggregates := range data {
+		for _, aggregate := range aggregates {
+			if aggregate.SourceFile() != fileURI {
+				continue
+			}
+
+			flattenedAggregates = append(flattenedAggregates, aggregate)
+		}
+	}
+
+	c.aggregateData[fileURI] = flattenedAggregates
+}
+
+func (c *Cache) GetFileAggregates(fileURI string) ([]report.Aggregate, bool) {
+	c.aggregateDataMu.Lock()
+	defer c.aggregateDataMu.Unlock()
+
+	data, ok := c.aggregateData[fileURI]
+	if !ok {
+		return nil, false
+	}
+
+	return data, true
```

Comment:
💭 Is that significantly different from this?
```go
return c.aggregateData[fileURI]
```

Reply:
Erm, no! Will fix, though there is plenty more to address here too 😅

Reply:
Hmm, that doesn't seem to work... 

```
internal/lsp/cache/cache.go:215:9: not enough return values
        have ([]report.Aggregate)
        want ([]report.Aggregate, bool)

```

Reply:
But this is fine:

```
	data, ok := c.aggregateData[fileURI]

	return data, ok
```

Reply:
Ah, right, it's not using the two-return value magic from simple returns. Well anything is fine, just nitpicking. I wasn't even completely sure if I had sent that comment or abandoned it 🙈

Reply:
all good! this PR has a way to go however, still!